### PR TITLE
fix: fix AllergyIntoleranceService lookup by pid

### PR DESCRIPTION
**Pull request by stephenwaite**
Fixes #

### CODE REVIEW

File: src/Services/AllergyIntoleranceService.php [hunk 1]
```diff
@@ -140,9 +142,11 @@ public function search($search, $isAndCondition = true)
      */
     public function getAll($search = array(), $isAndCondition = true, $puuidBind = null)
     {
-        // backwards compatible we let sub tables be referenced before, we want those to go away as it's a leaky abstraction
+        // backwards compatible we let sub tables be referenced before,
+        // we want those to go away as it's a leaky abstraction
         if (isset($search['lists.pid'])) {
-            $search['puuid'] = $search['lists.pid'];
+            $patient = (new PatientService())->findByPid($search['lists.pid']);
+            $search['puuid'] = UuidRegistry::uuidToString($patient['uuid']);
```
Comment (bradymiller) [inline src/Services/AllergyIntoleranceService.php:149]: nice!

@adunsulag , is there a chance this fix may unmask a bug and affect inferno testing?

Comment (adunsulag) [reply src/Services/AllergyIntoleranceService.php:149]: Yes @stephenwaite and @bradymiller, I'll need to look at this more closely as I thought we brought in a PR recently to 'supposedly' fix the very thing that @stephenwaite is fixing. I'm finding it odd that its still braking.  I've got some stuff going on today, but will see if I can get to it this evening or tomorrow.

Comment (adunsulag) [reply src/Services/AllergyIntoleranceService.php:149]: @stephenwaite I opened a PR to your PR with some changes. I thought the changes by @IvanJov 
 were in this same spot but it was lower down in the function.  I tested this and allergies are now showing up in the card. Sinces its hitting the lists.pid field for the search it shouldn't have an impact on the Inferno testing. 
 Here is the PR. https://github.com/stephenwaite/openemr/pull/9

**Comment by stephenwaite** [comment]
thanks @adunsulag and @bradymiller!

**Comment by bradymiller** [comment]
tenor99

**Pull request merged**